### PR TITLE
major changes

### PR DESCRIPTION
**Pull request by orhanobut**
- varargs added
```java
Logger.d("message %d %s", 23, "test");
```
- e(Exception e) is removed
- MethodCount and Tag definitions are moved from constructor to a new method 
```java
Logger.t(TAG, METHOD_COUNT).d(MESSAGE);
```
- bug fix for indexoutofbound
- new structure

### CODE REVIEW

**Comment by vincenthauser** [comment]
Great! This was one of the features I missed the most.

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+    @Override
+    public void e(String message, Object... args) {
+        e(null, message, args);
+    }
+
+    @Override
+    public void e(Throwable throwable, String message, Object... args) {
+        if (throwable != null && message != null) {
+            message += " : " + throwable.toString();
```
Comment (yekmer) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: Throwable toString method print is not meaningful for humans, changing it makes sense. it is same in 118th line too

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+    public void json(String json) {
+        String tag = getTag();
+        int methodCount = getMethodCount();
+
+        if (TextUtils.isEmpty(json)) {
+            d(tag, "Empty/Null json content", methodCount);
+            return;
+        }
+        try {
+            if (json.startsWith("{")) {
```
Comment (yekmer) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: json string can contain empty characters and still will be valid json, so startsWith should be changed with first non empty character method

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+                JSONObject jsonObject = new JSONObject(json);
+                String message = jsonObject.toString(JSON_INDENT);
+                d(tag, message, methodCount);
+                return;
+            }
+            if (json.startsWith("[")) {
```
Comment (yekmer) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: same here

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java (hunk 1 above)
Comment (orhanobut) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: actually it returns a meaningful message but this should be improved indeed. 
```java
    public String toString() {
        String s = getClass().getName();
        String message = getLocalizedMessage();
        return (message != null) ? (s + ": " + message) : s;
    }
```

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java (hunk 2 above)
Comment (orhanobut) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: for json validation, there should be another PR to cover everything

File: logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java (hunk 3 above)
Comment (orhanobut) [inline logger/src/main/java/com/orhanobut/logger/LoggerPrinter.java]: for json validation, there should be another PR to cover everything
